fe/fe-core/src/main/java/com/starrocks/catalog/TabletStatMgr.java

```diff
@@ -189,142 +195,183 @@ private void updateLocalTabletStat(Long beId, TTabletStatResult result) {
+            this.dbName = dbName;
+            this.tableName = tableName;
+            this.partitionId = partitionId;
+            this.visibleVersion = visibleVersion;
+            this.visibleVersionTime = visibleVersionTime;
+            this.tablets = Objects.requireNonNull(tablets);
+        }
+
+        private String debugName() {
+            return String.format("%s.%s.%d version %d", dbName, tableName, partitionId, visibleVersion);
+        }
+    }
+
+    private static class CollectTabletStatJob {
+        private final String dbName;
+        private final String tableName;
+        private final long partitionId;
+        private final long version;
+        private final Map<Long, Tablet> tablets;
+        private List<Future<TabletStatResponse>> responseList;
+
+        CollectTabletStatJob(PartitionSnapshot snapshot) {
+            this.dbName = Objects.requireNonNull(snapshot.dbName, "dbName is null");
+            this.tableName = Objects.requireNonNull(snapshot.tableName, "tableName is null");
+            this.partitionId = snapshot.partitionId;
+            this.version = snapshot.visibleVersion;
+            this.tablets = new HashMap<>();
+            for (Tablet tablet : snapshot.tablets) {
+                this.tablets.put(tablet.getId(), tablet);
             }
-        } catch (Throwable e) {
-            LOG.warn("failed to get lake tablet stats. table id: {}", table.getId(), e);
-            return;
         }
-        LOG.info("finished to get lake tablet stats. db id: {}, table id: {}, cost: {} ms", db.getId(), table.getId(),
-                (System.currentTimeMillis() - start));
 
-        if (idToStat.isEmpty()) {
-            return;
+        void execute() {
+            sendTasks();
+            waitResponse();
         }
 
-        // update tablet stats
-        locker.lockDatabase(db, LockType.WRITE);
-        try {
-            for (PhysicalPartition partition : table.getPhysicalPartitions()) {
-                long partitionId = partition.getId();
-                if (!partitionToVersion.containsKey(partitionId)) {
-                    continue;
+        private String debugName() {
+            return String.format("%s.%s.%d", dbName, tableName, partitionId);
+        }
+
+        private void sendTasks() {
+            Map<ComputeNode, List<TabletInfo>> beToTabletInfos = new HashMap<>();
+            for (Tablet tablet : tablets.values()) {
+                ComputeNode node = Utils.chooseNode((LakeTablet) tablet);
+                if (node == null) {
+                    LOG.warn("Stop sending tablet stat task for partition {} because no alive node", debugName());
+                    return;
                 }
+                TabletInfo tabletInfo = new TabletInfo();
+                tabletInfo.tabletId = tablet.getId();
+                tabletInfo.version = version;
+                beToTabletInfos.computeIfAbsent(node, k -> Lists.newArrayList()).add(tabletInfo);
```

Comment:
what if the node exists in the map? the tabletInfo will be abandoned?

Reply:
The tabletInfo will be added to the `List<TabletInfo>`

Reply:
List<TabletInfo> infos = beToTabletInfos.computeIfAbsent(node, k -> Lists.newArrayList());
infos.add(tabletInfo);